pkgs/development/python-modules/scs/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, pythonOlder
+, buildPythonPackage
+, fetchFromGitHub
+, liblapack
+, numpy
+, blas
+, scipy
+  # check inputs
+, nose
+}:
+
+buildPythonPackage rec {
+  pname = "scs";
+  version = "2.1.1";
+
+  disabled = pythonOlder "3.5";
+
+  src = fetchFromGitHub {
+    owner = "bodono";
+    repo = "scs-python";
+    rev = "f02abdc0e2e0a5851464e30f6766ccdbb19d73f0"; # need to choose commit manually, untagged
+    sha256 = "01ghvyylxql7jvdcwy65wf7y5qykqyqkbysy402zfhgmm9dxizgv";
+    fetchSubmodules = true;
```

Comment:
Ihttps://github.com/bodono/scs-python/blob/master/setup.py
If you look at what the upstream src repo and `setup.py` are doing, it appears they're shipping the `scs` C library as a git submodule, then having python recompile it from scratch.

In the nix distro, we generally prefer to have the C package stand-alone, then have the python package just link the package as a proper distribution dependency without vendoring.  An example of a package that does this is pyarrow:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/pyarrow/default.nix

Reply:
I'm trying to do this, and I'm stuck with getting the proper nix path. I have the following:
```
prePatch = ''
    cp -r ${scs.src} ./scs
'';
```
Issue is that the actual source is stored at something like ``${scs.src}/j4i4blqll2d3wrzzllh1b2hz52s2iybs-source``, based on the derivation name/hash, I believe. Any tips on how to get this path?

Note that this is different from ``pyarrow`` b/c that includes the python source in the same directory.

Reply:
`${scs.src}` should work; you can view it like this:
```
λ brh nixpkgs →  tree $(nix-build -A scs.src)
/nix/store/q1alada06jhjpk7cnqjsy92pdlzpyj6n-source
├── include
│   ├── accel.h
│   ├── cones.h
...
```

You probably have to rename this one to `python-scs` to avoid infinite recursion on the `scs` name.  
https://github.com/bhipple/nixpkgs/commit/494f2efd434e8d87623b93c7b73d61c9935d8b44

In an ideal world, we would rely on just `${scs}` (the already-compiled binary package with SO) in the python package, rather than the src directly, and re-use the SO's and headers from the installed `${scs}`.  Doing so may require more patching to the setup.py, but I'd give it a shot to see if it isn't too hard, as that's the "best" solution.  In lieu of that, vendoring the binaries but sharing the src is second best, and if THAT doesn't work then we're back to the initial PR ... with the knowledge that these things may not stay in sync :)

Reply:
When I do all of the above, I find:

```
src/scsmodule.c: In function ‘csolve’:
src/scsmodule.c:331:20: error: ‘ScsSettings’ {aka ‘struct SCS_SETTINGS’} has no member named ‘write_data_filename’
  331 |           &(d->stgs->write_data_filename),
      |                    ^~
```
which suggests that our version of `scs` is not in sync with what `python-scs` expects?

Reply:
Thanks for the help & examples. I'll commit the above and try to get the code/build working. ^scs version might be due to not having #78896 in your branch?